### PR TITLE
Base64 using query attribute

### PR DESCRIPTION
**Pull request by ashtul**

### CODE REVIEW

File: src/vector_index.h [hunk 1]
```diff
@@ -17,6 +17,9 @@
 #define VECSIM_M "M"
 #define VECSIM_EF "EF"
 
+#define BASE64_ENABLED 1
+#define BASE64_DISABLED 2
+
```
Comment (rafie) [inline src/vector_index.h:22]: other values possible? is this a bool?

File: src/vector_index.h [hunk 2]
```diff
@@ -27,6 +30,7 @@ typedef struct VectorFilter {
   void *vector;                   // vector data
   size_t vecLen;                  // vector length
   VectorQueryType type;           // TOPK or RANGE
+  int isBase64;                   // uses base64 strings
```
Comment (rafie) [inline src/vector_index.h:33]: bool?

File: src/vector_index.c [hunk 3]
```diff
@@ -71,15 +71,18 @@ IndexIterator *NewVectorIterator(RedisSearchCtx *ctx, VectorFilter *vf) {
   unsigned char *vector = vf->vector;
   switch (vf->type) {
     case VECTOR_TOPK:
+      if (vf->isBase64) {
+        vector = base64_decode(vf->vector, vf->vecLen, &outLen);
+        // TODO: check outLen == expected len.
+        unescape((char *)vector, strlen((char *)vector));
```
Comment (rafie) [inline src/vector_index.c:77]: why are we unescaping here?

File: src/vector_index.h (hunk 1 above)
Comment (ashtul) [reply src/vector_index.h:22]: Yes.
For the Phonetic attribute, there are `define`s so I followed in the same footsteps.

File: src/vector_index.h (hunk 2 above)
Comment (ashtul) [reply src/vector_index.h:33]: Generally, we do not use bool in RediSearch. I guess it follows Redis OSS.

File: src/vector_index.c (hunk 3 above)
Comment (ashtul) [reply src/vector_index.c:77]: We have to unescape for `+` and `/` which are part of the `BASE64` standard.

File: src/vector_index.h (hunk 2 above)
Comment (rafie) [reply src/vector_index.h:33]: But we do. Just search `bool` on the code base. I'm not familiar with such a convention.

File: src/vector_index.h (hunk 1 above)
Comment (rafie) [reply src/vector_index.h:22]: I meant, logically. Can it have other values?

File: src/vector_index.c (hunk 3 above)
Comment (rafie) [reply src/vector_index.c:77]: Isn't it is only when initially escaping + and / in URL context? Like, when taking a binary buffer (which is not even text), does this have meaning at all?

File: src/vector_index.h (hunk 1 above)
Comment (ashtul) [reply src/vector_index.h:22]: Not that I can think of. We might opt out for a different solution than base64 but for now, this is the path we chose.

File: src/vector_index.c (hunk 3 above)
Comment (ashtul) [reply src/vector_index.c:77]: Any string with these characters going through the parser will get an error message if not escape.

**Review by rafie — APPROVED** [review]

**Pull request merged**